src/components/auth/Registration.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react';
+import {
+  Box,
+  Button,
+  Link,
+  TextField,
+} from '@mui/material';
+import Header from '../../elements/Header';
+import Logo from '../../elements/Logo';
+import './Auth.scss';
+
+const Registration = () => {
+  const [login, setLogin] = useState('');
+  const [password, setPassword] = useState('');
+  const [retryPassword, setRetryPassword] = useState('');
+  const [errorAlertLog, setErrorAlertLog] = useState(false);
+  const [errorAlertPass, setErrorAlertPass] = useState(false);
+  const [errorAlertPassReplay, setErrorAlertPassReplay] = useState(false);
+  const [errorText, setErrorText] = useState('');
+
+  const formSubmit = () => {
+    if (!(login || password || retryPassword)) {
+      setErrorText('Введите данные');
+      setErrorAlertLog(true);
+      setErrorAlertPass(true);
+      setErrorAlertPassReplay(true);
+    } else {
+      setErrorAlertLog(false);
+      setErrorAlertPass(false);
+      setErrorAlertPassReplay(false);
+      setErrorText('');
+      if (login.length < 6 || (!/[a-z]/.test(login))) {
+        setErrorText('Логин должен состоять из латинского алфавита и длиной не менее 6 символов');
+        setErrorAlertLog(true);
+        setErrorAlertPass(true);
+        setErrorAlertPassReplay(true);
+      } else {
+        setErrorAlertLog(false);
+        const regExp = /^(?=.*[A-Za-z])(?=.*\d)[a-zA-Z0-9!@#$%^&*()~¥=_+}{":;'?/>.<,`\-\|\[\]]{6,20}$/;
+        if (!regExp.test(password)) {
+          setErrorAlertPass(true);
+          setErrorText('Некорректный пароль');
+        } else {
+          setErrorAlertPass(false);
+          setErrorText('');
+          if (!(password === retryPassword)) {
+            setErrorAlertPassReplay(true);
+            setErrorAlertPass(true);
+            setErrorText('Пароли не совпадают');
+          } else {
+            setErrorAlertPassReplay(false);
+            setErrorText('');
+            console.log('AXIOSSSSS!!!!');
```

Comment:
!!!

Reply:
sorry, this `console.log` helped check validation form. this is temporarily. but? i deleted log now